component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/RecordImpl.java

```diff
@@ -115,25 +115,37 @@ public BuilderImpl(final Schema providedSchema) {
             this.providedSchema = providedSchema;
         }
 
-        private void validateTypeAgainstProvidedSchema(final String name, final Schema.Type type, final Object value) {
-            if (providedSchema == null) {
-                return;
-            }
-            if (entryIndex == null) {
-                entryIndex = providedSchema.getEntries().stream().collect(toMap(Schema.Entry::getName, identity()));
+        private Schema.Entry findMandatoryEntry(final String name) {
+            assert this.providedSchema != null;
```

Comment:
why using `assert` here?

Reply:
this function is private and must be called only when providedSchema exists. assert is normally not tested on running code, just in unit test (until we used -ea option)

Reply:
still think that its place in in tests not here...

Reply:
assert removed